neural_compressor/transformers/models/modeling_auto.py

```diff
@@ -0,0 +1,644 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# coding=utf-8
+# Copyright 2021 The EleutherAI and HuggingFace Teams. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import os
+import types
+
+from accelerate import init_empty_weights
+
+from neural_compressor.adaptor.torch_utils.util import set_module
+from neural_compressor.torch.algorithms.weight_only.modules import INCWeightOnlyLinear
+from neural_compressor.transformers import GPTQConfig, RtnConfig
+from neural_compressor.transformers.quantization.utils import convert_dtype_torch2str, replace_linear, save_low_bit
+from neural_compressor.utils import logger
+from neural_compressor.utils.utility import CpuInfo, LazyImport
+
+torch = LazyImport("torch")
+transformers = LazyImport("transformers")
+transformers_configuration_utils = LazyImport("transformers.configuration_utils")
+
+
+def build_woq_model(model, quantization_config):
+    bits = quantization_config.bits
+    for n, m in model.named_modules():
+        if n in quantization_config.modules_to_not_convert:
+            continue
+        if isinstance(m, torch.nn.Linear):
+            zp = getattr(
+                quantization_config,
+                "zero_point",
+                not getattr(quantization_config, "sym", False),
+            )
+            use_optimum_format = True
+            with init_empty_weights():
+                new_module = INCWeightOnlyLinear(
+                    m.in_features,
+                    m.out_features,
+                    dtype="int4" if bits == 4 else "int8",
+                    bits=quantization_config.bits,
+                    group_size=quantization_config.group_size,
+                    zp=zp,
+                    bias=m.bias is not None,
+                    g_idx=True,
+                    use_optimum_format=use_optimum_format,
+                )
+            set_module(model, n, new_module)
+    return model
+
+
+class _BaseQBitsAutoModelClass:
```

Comment:
do we still need support for QBit? or here is just reused the original name from ITREX but no real connection with QBits?  if so, suggest to change to a proper name

Reply:
yes, it is just reuse the original name, improve it to _BaseINCAutoModelClass.